message UisInputData {
	required uint32 type = 1;
	required uint32 count = 2;
	required bytes  body = 3;
}

message UisPkt {
	required uint32 sourceId = 1;
	required uint32 moduleId = 2;
	required uint32 seq      = 3;

	enum PktCmd {
		INIT = 0x10000001;
		OPEN = 0x10000002;         // open device
		CLOSE = 0x10000003;        // close deivce
		CLOSE_SOURCE = 0x10000004; // close source 
		IOCTL = 0x10000005;        // ioctl command

		INPUT = 0x00000001;
	}

	required PktCmd cmd      = 4; 

//	required uint32 type = 5;
//	required uint32 count = 6;
//	required bytes  body = 7;

	required UisInputData data = 10;
}

service UisService {
  rpc Open  (UisPkt) returns (UisPkt);
  rpc Ioctl (UisPkt) returns (UisPkt);
}
